AttributeError: 'FloatObject' object has no attribute 'items'

Starting with **1.27.10** `PyPDF2.PdfFileReader(file_object, strict=False)` will cause a trace like:
```
.venv/lib/python3.7/site-packages/PyPDF2/_reader.py:239: in __init__
    self.read(stream)
.venv/lib/python3.7/site-packages/PyPDF2/_reader.py:965: in read
    self._rebuild_xref_table(stream)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    def _rebuild_xref_table(self, stream):
        ...
        # code below duplicated
        readNonWhitespace(stream)
        stream.seek(-1, 1)
        new_trailer = readObject(stream, self)
>       for key, value in list(new_trailer.items()):
E       AttributeError: 'FloatObject' object has no attribute 'items'
```
when reading PDFs (attached below) with xref trailer issues which require rebuilding. And this reading worked like a charm before! (**1.27.9** and lower)

## Environment

Replicated it on Linux and Mac.

## Code

_mypdf.py_
```python
import PyPDF2

with open("vero.pdf", "rb") as file_object:
    print(PyPDF2.PdfFileReader(file_object, strict=False).getNumPages())
```

```
% python mypdf.py
PdfReadWarning: incorrect startxref pointer(4) [_reader.py:938]
Traceback (most recent call last):
  File "/Users/cmin/Repos/rpaframework/packages/pdf/mypdf.py", line 4, in <module>
    print(PyPDF2.PdfFileReader(file_object, strict=False).getNumPages())
  File "/Users/cmin/Repos/rpaframework/packages/pdf/.venv/lib/python3.9/site-packages/PyPDF2/_reader.py", line 239, in __init__
    self.read(stream)
  File "/Users/cmin/Repos/rpaframework/packages/pdf/.venv/lib/python3.9/site-packages/PyPDF2/_reader.py", line 963, in read
    self._rebuild_xref_table(stream)
  File "/Users/cmin/Repos/rpaframework/packages/pdf/.venv/lib/python3.9/site-packages/PyPDF2/_reader.py", line 1199, in _rebuild_xref_table
    for key, value in list(new_trailer.items()):
AttributeError: 'FloatObject' object has no attribute 'items'
```

## PDF

[vero.pdf](https://github.com/py-pdf/PyPDF2/files/8603317/vero.pdf)

--- a/PyPDF2/_reader.py
+++ b/PyPDF2/_reader.py
@@ -962,8 +962,11 @@ class PdfFileReader(object):
                 else:
                     break
             elif xref_issue_nr:
-                self._rebuild_xref_table(stream)
-                break
+                try:
+                    self._rebuild_xref_table(stream)
+                    break
+                except Exception:
+                    xref_issue_nr = 0
             elif x.isdigit():
                 xrefstream = self._read_pdf15_xref_stream(stream)
 
@@ -1197,7 +1200,10 @@ class PdfFileReader(object):
         # code below duplicated
         readNonWhitespace(stream)
         stream.seek(-1, 1)
+
+        # there might be something that is not a dict (see #856)
         new_trailer = readObject(stream, self)
+
         for key, value in list(new_trailer.items()):
             if key not in self.trailer:
                 self.trailer[key] = value

--- a/PyPDF2/_reader.py
+++ b/PyPDF2/_reader.py
@@ -1176,11 +1176,11 @@ class PdfFileReader(object):
             line += stream.read(2)  # 1 char already read, +2 to check "obj"
             if line.lower() != b_("obj"):
                 return 3
-            while stream.read(1) in b_(" \t\r\n"):
-                pass
-            line = stream.read(256)  # check that it is xref obj
-            if b_("/xref") not in line.lower():
-                return 4
+            # while stream.read(1) in b_(" \t\r\n"):
+            #     pass
+            # line = stream.read(256)  # check that it is xref obj
+            # if b_("/xref") not in line.lower():
+            #     return 4
         return 0
 
     def _rebuild_xref_table(self, stream):

